src/zc/buildout/__init__.py

```diff
@@ -13,9 +13,108 @@
 ##############################################################################
 """Buildout package
 """
+from setuptools.extern.six.moves import urllib
+from setuptools.package_index import PackageIndex
+from setuptools.package_index import URL_SCHEME
+from setuptools.package_index import HREF
+from setuptools.package_index import distros_for_url
+from setuptools.package_index import htmldecode
+
+from pip._internal.index.collector import HTMLPage
+from pip._internal.index.collector import parse_links
+from pip._internal.index.package_finder import _check_link_requires_python
+from pip._internal.models.target_python import TargetPython
+from pip._vendor import six
+
+PY_VERSION_INFO = TargetPython().py_version_info
+
+
+def patch_PackageIndex():
+    setattr(PackageIndex, 'process_url', process_url)
+
+
+def process_url(self, url, retrieve=False):
+    """Evaluate a URL as a possible download, and maybe retrieve it"""
+    if url in self.scanned_urls and not retrieve:
+        return
+    self.scanned_urls[url] = True
+    if not URL_SCHEME(url):
+        self.process_filename(url)
+        return
+    else:
+        dists = list(distros_for_url(url))
+        if dists:
+            if not self.url_ok(url):
+                return
+            self.debug("Found link: %s", url)
+
+    if dists or not retrieve or url in self.fetched_urls:
+        list(map(self.add, dists))
```

Comment:
```suggestion
        for dist in dists:
            self.add(dist)
```

Reply:
This is code stolen from setuptools as I am monkeypatching.

https://github.com/pypa/setuptools/blob/v44.1.0/setuptools/package_index.py#L338

I decided to not touch what does not need to be touched. What is your take on this ?

Reply:
If there's reason to believe in the future someone may want to merge the copy-pasted code with latest upstream changes, then keeping our changes to the minimum is the right thing.

I didn't realize this was copypasta.